### PR TITLE
Update heroku deploy docs

### PR DESCRIPTION
**Pull request by chap**
## Hey, I just made a Pull Request!

Updated Heroku deploy documentation.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/deployment/heroku.md [hunk 1]
```diff
@@ -5,69 +5,102 @@ sidebar_label: Heroku
...
-deployment in a hands-off way. Heroku supports container deployment of Docker
-images, a natural fit for Backstage.
+Heroku is a Platform as a Service (PaaS) designed to simplify application deployment.
 
-## Configuring the CLI
+## Create App
 
-First, install the
-[heroku-cli](https://devcenter.heroku.com/articles/heroku-cli) and login:
+Start with an exisiting Backstage app or follow the [getting started guide](https://backstage.io/docs/getting-started/) to create a new one:
```
Comment (awanlin) [inline docs/deployment/heroku.md:12]:
```suggestion
Starting with an existing Backstage app or follow the [getting started guide](https://backstage.io/docs/getting-started/) to create a new one:
```

File: docs/deployment/heroku.md [hunk 2]
```diff
@@ -5,69 +5,102 @@ sidebar_label: Heroku
...
       # https://devcenter.heroku.com/articles/dynos#web-dynos
 ```
 
-> Make sure your file is being copied into your container in the `Dockerfile`.
+## Build Script
+
+Add a build script in `package.json` to compile frontend during deployment:
+```json
+"scripts": {
+  "build": "yarn build:backend --config ../../app-config.yaml"
```
Comment (awanlin) [inline docs/deployment/heroku.md:54]:
```suggestion
  "build": "yarn build:backend --config ../../app-config.yaml --config app-config.production.yaml"
```

File: docs/deployment/heroku.md [hunk 3]
```diff
@@ -5,69 +5,102 @@ sidebar_label: Heroku
...
+```
+
+## Start Command
```
Comment (awanlin) [inline docs/deployment/heroku.md:57]: Is there a reason we aren't suggesting running in a Docker container? We do this right now with the Demo site. I think perhaps we should keep both methods.

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for taking this on @chap, left some comments, the biggest one I have is around running a container.

File: docs/deployment/heroku.md (hunk 3 above)
Comment (chap) [reply docs/deployment/heroku.md:52]: I have added the docker instructions.

**Comment by chap** [comment]
@awanlin thank you for the feedback.

I think Heroku's git deploys are better for most folks, because it doesn't require any local build commands and enables git sync features.

However, I have added the docker commands back to this doc as an alternative for those that prefer it.

Please let me know if there are any more suggestions.

Thanks again!

**Comment by chap** [comment]
@awanlin just added a "Deploy to Heroku" [button](https://devcenter.heroku.com/articles/heroku-button) that will bootstrap an app from `backstage/create-app@latest`.

[Here's the source repo](https://github.com/chap/backstage-button), had some fun with the logos :)

image

**Comment by awanlin** [comment]
Hi @chap, thanks for the updates, plan is to review this on Friday. As for the button I don't think this is something we would want to add, sorry. 

@Rugvip what's your thoughts on this button?

**Comment by chap** [comment]
@awanlin no problem if we want to drop the button. thanks!

File: docs/deployment/heroku.md [hunk 4]
```diff
@@ -5,69 +5,138 @@ sidebar_label: Heroku
 description: How to deploy Backstage to Heroku
 ---
 
-Heroku is a Platform as a Service (PaaS) designed to handle application
-deployment in a hands-off way. Heroku supports container deployment of Docker
-images, a natural fit for Backstage.
+Heroku is a Platform as a Service (PaaS) designed to simplify application deployment.
 
-## Configuring the CLI
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/chap/backstage-button)
```
Comment (awanlin) [inline docs/deployment/heroku.md:10]: If we can remove this that would be ideal, just leaves us open to something going unmaintained that can be easily missed.

File: docs/deployment/heroku.md [hunk 5]
```diff
@@ -5,69 +5,138 @@ sidebar_label: Heroku
...
 ```
 
-## Push and deploy a Docker image
+Configure the Heroku app to run a container image:
+
+```shell
+$ heroku stack:set container -a <your-app>
+```
+
+Locally run the [host build command](https://backstage.io/docs/deployment/docker/#host-build):
```
Comment (awanlin) [inline docs/deployment/heroku.md:128]:
```suggestion
Locally run the [host build commands](https://backstage.io/docs/deployment/docker/#host-build), they must be run whenever you are going to publish a new image:
```

File: docs/deployment/heroku.md [hunk 6]
```diff
@@ -5,69 +5,138 @@ sidebar_label: Heroku
...
+
+```shell
+$ yarn build:backend --config ../../app-config.yaml --config ../../app-config.production.yaml
+```
 
-Now we can push a Backstage [Docker image](docker.md) to Heroku's container
-registry and release it to the `web` worker:
+Build, push, and release the container image:
```
Comment (awanlin) [inline docs/deployment/heroku.md:134]:
```suggestion
Build, push, and release the container image to the `web` worker:
```

File: docs/deployment/heroku.md [hunk 7]
```diff
@@ -5,69 +5,138 @@ sidebar_label: Heroku
...
+Configure the Heroku app to run a container image:
+
+```shell
+$ heroku stack:set container -a <your-app>
+```
+
+Locally run the [host build command](https://backstage.io/docs/deployment/docker/#host-build):
+
+```shell
+$ yarn build:backend --config ../../app-config.yaml --config ../../app-config.production.yaml
```
Comment (awanlin) [inline docs/deployment/heroku.md:131]:
```suggestion
yarn install --frozen-lockfile
yarn tsc
yarn build:backend --config ../../app-config.yaml --config ../../app-config.production.yaml
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Looking good @chap, just some small adjustments needed.

One big ask, but simple to to, can you remove the `$` from all the commands? I know it was there before but when you use the copy feature that gets included and then the command fails. 👍

File: docs/deployment/heroku.md (hunk 4 above)
Comment (chap) [reply docs/deployment/heroku.md]: No problem, removed

File: docs/deployment/heroku.md (hunk 5 above)
Comment (chap) [reply docs/deployment/heroku.md]: fixed

File: docs/deployment/heroku.md (hunk 7 above)
Comment (chap) [reply docs/deployment/heroku.md]: fixed

**Comment by chap** [comment]
@awanlin thanks for your feedback.

I've made all the suggested changes.

**Comment by awanlin** [comment]
Hi @benjdlambert, ready for you to review, just docs changes 👍

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :tada:

**Pull request merged**